DigitalLearningSolutions.Data/DataServices/SelfAssessmentDataService/SelfAssessmentSupervisorDataService.cs

```diff
@@ -0,0 +1,378 @@
+            );
+
+            if (numberOfAffectedRows < 1)
+            {
+                logger.LogWarning(
+                    "Not inserting CandidateAssessmentSupervisorVerification as db update failed. " +
+                    $"candidateAssessmentSupervisorId: {candidateAssessmentSupervisorId}"
+                );
+            }
+        }
+
+        public void UpdateCandidateAssessmentSupervisorVerificationEmailSent(
+            int candidateAssessmentSupervisorVerificationId
+        )
+        {
+            connection.Execute(
+                @"UPDATE CandidateAssessmentSupervisorVerifications
+                    SET EmailSent = GETUTCDATE()
+                    WHERE ID = @candidateAssessmentSupervisorVerificationId",
+                new { candidateAssessmentSupervisorVerificationId }
+            );
+        }
+
+        public SupervisorComment? GetSupervisorComments(int candidateId, int resultId)
+        {
+            return connection.Query<SupervisorComment>(
+                @"SELECT
+                        sar.AssessmentQuestionID, sea.Name, sasv.Comments, sar.CandidateID, sar.CompetencyID, com.Name as CompetencyName,
+                        sar.SelfAssessmentID, sasv.CandidateAssessmentSupervisorID, sasv.SelfAssessmentResultId, sasv.Verified,
+                        sar.ID, sstrc.CompetencyGroupID, sea.Vocabulary, sasv.SignedOff
+                    FROM SelfAssessmentResultSupervisorVerifications AS sasv
+                    INNER JOIN SelfAssessmentResults AS sar
+                        ON sasv.SelfAssessmentResultId = sar.ID
+                    INNER JOIN SelfAssessments AS sea
+                        ON sar.SelfAssessmentID = sea.ID
+                    INNER JOIN SelfAssessmentStructure AS sstrc
+                        ON sar.CompetencyID = sstrc.CompetencyID 
+                    INNER JOIN Competencies AS com
+                        ON sar.CompetencyID = com.ID
+                    WHERE (sar.CandidateID = @candidateId) AND (sasv.SelfAssessmentResultId = @resultId)",
+                new { candidateId, resultId }
+            ).FirstOrDefault();
+        }
+
+        public IEnumerable<Administrator> GetValidSupervisorsForActivity(
+            int centreId,
+            int selfAssessmentId,
+            int candidateId
+        )
+        {
+            return connection.Query<Administrator>(
+                @"SELECT
+                        AdminID,
+                       Forename,
+                       Surname,
+                       Email,
+                       ProfileImage,
+                       IsFrameworkDeveloper
+                    FROM AdminUsers
+                    WHERE (
+                        (Supervisor = 1) AND (Active = 1) AND (CategoryID = 0) AND (CentreID = @centreId)
+                        OR
+                        (Supervisor = 1) AND (Active = 1) AND (CategoryID = (SELECT CategoryID FROM SelfAssessments WHERE (ID = @selfAssessmentId))) AND (CentreID = @centreId)

```

Comment:
Don't know how much cleaning up of the SQL you were aiming to do, but there are a number old queries that have repeated conditions that could be simplified. But that might be too much faff, especially with the lack of unit tests.

Reply:
Yeah, I don't want to be changing any SQL queries other than formatting them otherwise we'll have to re-test properly.